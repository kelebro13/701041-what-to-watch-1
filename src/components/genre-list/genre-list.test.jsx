import renderer from "react-test-renderer";
import GenreList from "./genre-list";

it(`render correctly GenreList component`, () => {
  const genres = [`All genres`, `Comedies`, `Crime`, `Documentary`, `Dramas`, `Horror`, `Kids & Family`, `Romance`, `Sci-Fi`, `Thrillers`];
  const activeGenre = `All genres`;
  const onSelectedGenreChange = () => {};

  const tree = renderer
    .create(<GenreList genres={genres} activeGenre={activeGenre} onSelectedGenreChange={onSelectedGenreChange}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
