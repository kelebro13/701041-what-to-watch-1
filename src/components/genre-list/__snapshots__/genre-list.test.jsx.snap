// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly GenreList component 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
    >
      All genres
    </button>
  </li>
  <li
    className="catalog__genres-item"
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Comedies
    </button>
  </li>
  <li
    className="catalog__genres-item"
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Crime
    </button>
  </li>
  <li
    className="catalog__genres-item"
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Documentary
    </button>
  </li>
  <li
    className="catalog__genres-item"
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Dramas
    </button>
  </li>
  <li
    className="catalog__genres-item"
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Horror
    </button>
  </li>
  <li
    className="catalog__genres-item"
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Kids & Family
    </button>
  </li>
  <li
    className="catalog__genres-item"
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Romance
    </button>
  </li>
  <li
    className="catalog__genres-item"
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Sci-Fi
    </button>
  </li>
  <li
    className="catalog__genres-item"
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Thrillers
    </button>
  </li>
</ul>
`;
